irodsManager.py

```diff
@@ -518,3 +532,1757 @@ def touch(self, lpath, seconds_since_epoch=-1, reference=''):
+        # returns
+        # - Status code and response message.
+        def modify_metadata(self, lpath: str, operations: list, admin: int=0):
+            if (self.token == None):
+                raise Exception('No token set. Use setToken() to set the auth token to be used')
+            if (not isinstance(lpath, str)):
+                raise Exception('lpath must be a string')
+            if (not isinstance(operations, list)):
+                raise Exception('operations must be a list of dictionaries')
+            if (not isinstance(operations[0], dict)):
+                raise Exception('operations must be a list of dictionaries')
+            if ((not admin == 0) and (not admin == 1)):
+                raise Exception('admin must be an int 1 or 0')
+
+            headers = {
+                'Authorization': 'Bearer ' + self.token,
+                'Content-Type': 'application/x-www-form-urlencoded',
+            }
+
+            data = {
+                'op': 'modify_metadata',
+                'lpath': lpath,
+                'operations': json.dumps(operations),
+                'admin': admin
+            }
+
+            r = requests.post(self.url_base + '/data-objects', headers=headers, data=data)
+
+            if (r.status_code / 100 == 2):
+                rdict = r.json()
+
+                if rdict['irods_response']['status_code']:
+                    print('Failed to modify metadata for \'' + lpath + '\': iRODS Status Code' + str(rdict['irods_response']['status_code']))
+                else:
+                    print('Metadata for \'' + lpath + '\' modified successfully')
+                
+                return(rdict)
+            else:
+                print('Error: ' + r.text)
+
+                return(rdict)
+            
+
+        # Sets the permission of a user for a given collection
+        # params
+        # - lpath: The absolute logical path of the collection to have a permission set.
+        # - entity_name: The name of the user or group having its permission set.
+        # - permission: The permission level being set. Either 'null', 'read', 'write', or 'own'.
+        # - admin (optional): Set to 1 to run this operation as an admin, otherwise set to 0. Defaults to 0.
+        # returns
+        # - Status code and response message.
+        def set_permission(self, lpath: str, entity_name: str, permission: str, admin: int=0):
+            if (self.token == None):
+                raise Exception('No token set. Use setToken() to set the auth token to be used')
+            if (not isinstance(lpath, str)):
+                raise Exception('lpath must be a string')
+            if (not isinstance(entity_name, str)):
+                raise Exception('entity_name must be a string')
+            if (not isinstance(permission, str)):
+                raise Exception('permission must be a string (\'null\', \'read\', \'write\', or \'own\')')

```

Comment:
i don't think we're limited to these four... should be 10... @korydraughn ?

Reply:
`set_permission` accepts the full range of permissions. See https://github.com/irods/irods_client_http_api/blob/f524cec80187a003bf2956d7890b26b68b1d86c8/core/src/common.cpp#L581-L601.

`modify_permissions` only accepts the original four permissions.

Reply:
oh.  really?   why the limitation to four for `modify`?

Reply:
The atomic API has not been updated to support the other levels.
- https://github.com/irods/irods/blob/7b8b97a1966bce4f18bd863997ca977a07485e5b/plugins/api/src/atomic_apply_acl_operations.cpp#L191-L198
- https://github.com/irods/irods/blob/7b8b97a1966bce4f18bd863997ca977a07485e5b/plugins/api/src/atomic_apply_acl_operations.cpp#L207-L217

Reply:
i see.  okay, let's make a new issue for that in the main repo if we don't already have one.    as it's already causing 'ripples' out here.

Reply:
Created https://github.com/irods/irods/issues/7913.

Reply:
thanks - resolving.